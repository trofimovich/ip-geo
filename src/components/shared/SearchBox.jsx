import React, { useState } from 'react';
import { string, func } from 'prop-types';

const SearchBox = ({ placeholder, btnText, onSearch }) => {
  const [searchString, setSearchString] = useState('');
  return (
    <div className="input-group mb-3 search-box">
      <input
        type="text"
        className="form-control"
        placeholder={placeholder}
        onChange={(evt) => setSearchString(evt)}
        aria-describedby="button-addon2"
      />
      <button
        onClick={() => onSearch(searchString)}
        className="btn btn-primary"
        type="submit"
        id="button-addon2"
      >
        {btnText}
      </button>
    </div>
  );
};

export default SearchBox;

SearchBox.defaultProps = {
  placeholder: 'Search',
  btnText: 'Search',
};

SearchBox.propTypes = {
  placeholder: string,
  btnText: string,
  onSearch: func.isRequired,
};
