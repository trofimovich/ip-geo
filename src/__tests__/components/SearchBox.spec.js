import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import SearchBox from '../../components/shared/SearchBox';

const noop = () => {};

describe('SearchBox', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(
      <SearchBox onSearch={noop} />,
    );
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Renders correctly with params', () => {
    const wrapper = shallow(
      <SearchBox
        onSearch={noop}
        placeholder="Enter some...."
        btnText="Test"
        validationMessage="Some validation message"
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should provide searchString to callback after search button was pressed', () => {
    const onSearch = jest.fn();

    const wrapper = mount(<SearchBox onSearch={onSearch} />);
    const searchInput = wrapper.find('input.form-control');
    searchInput.simulate('change', { target: { value: 'habr.com' }});

    const searchButton = wrapper.find('button');
    searchButton.simulate('click');

    expect(onSearch).toHaveBeenCalledWith('habr.com');

    wrapper.unmount();
  });

  it('Should invoke validate function', () => {
    const validate = jest.fn();

    const wrapper = mount(<SearchBox onSearch={noop} validate={validate} />);
    const searchInput = wrapper.find('input.form-control');
    searchInput.simulate('change', { target: { value: '1.1.1.1' }});
    searchInput.simulate('blur');
    expect(validate).toHaveBeenCalledWith('1.1.1.1');

    wrapper.unmount();
  })
});
