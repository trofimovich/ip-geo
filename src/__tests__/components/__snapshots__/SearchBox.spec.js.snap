// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Renders correctly 1`] = `
<div
  className="input-group mb-3 search-box"
>
  <input
    aria-describedby="button-addon2"
    className="form-control "
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Search"
    type="text"
  />
  <button
    className="btn btn-primary"
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    Search
  </button>
  <div
    className="invalid-feedback"
  >
    Field has incorrect value
  </div>
</div>
`;

exports[`SearchBox Renders correctly with params 1`] = `
<div
  className="input-group mb-3 search-box"
>
  <input
    aria-describedby="button-addon2"
    className="form-control "
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Enter some...."
    type="text"
  />
  <button
    className="btn btn-primary"
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    Test
  </button>
  <div
    className="invalid-feedback"
  >
    Some validation message
  </div>
</div>
`;
